bgpd/bgp_clist.c

```diff
@@ -311,9 +311,9 @@ static void community_list_delete(struct community_list_master *cm,
 	community_list_free(list);
 }
 
-static int community_list_empty_p(struct community_list *list)
+static bool community_list_empty_p(struct community_list *list)
 {
-	return (list->head == NULL && list->tail == NULL) ? 1 : 0;
+	return (list->head == NULL && list->tail == NULL) ? true : false;
```

Comment:
list->head == NULL && list->tail == NULL is already returning you boolean value which is true or false. Suggest to change it to 
`return list->head == NULL && list->tail == NULL`